proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,107 @@
+As outlined in the customer request on this GitHub [issue](https://github.com/NuGet/Home/issues/6060), there is a call for the incorporation of `nuget list` functionality into dotnet. The customers are requesting for this feature, but the `search` command can replicate the same functionality, with the only difference being the quantity of package results it outputs. The search command mitigates unnecessary server use by limiting the number of results through the `--take` option. By integrating this search functionality, we aim to accommodate our customers' needs, especially for scripting purposes.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+Imagine you have your dotnet CLI open. You would like to look up a NuGet package named `MyPackage` from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search MyPackage --source <MySource>`. It will provide you with the list of the packages in the source `<MySource>` that match with the search criteria.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options
+
+| Option | Function |
+|---------|:----------|
+| `--source` | A list of packages to search |
+| `--exact-match` | Return exact matches only as a search result |
+| `--prerelease` | Allow prerelease packages to be shown. |
+| `--non-interactive` | Do not prompt for user input or confirmations.|
+| `--take` | The number of results to return. The default value is 20.|
+| `--help` | Show command help and usage information |
+| `--verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+|||
+
+#### **Algorithm**
+
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--source`, `[Search term]`, `--prerelease` values.
+3. Based on `--verbosity` value, provide with a list of packages with various verbosity
+
+- Verbosity
+  - **Quiet** : Each line would look as follows :
+
+            >[Package Name] | [Latest Package Version]
+  - **Normal** :
+
+                >[Package Name] | [Latest Package Version] | [Amount of Downloads]
+                <Description of the Package(short form)>
+  - **Detailed** :
+
+                >[Package Name] | [Latest Package Version] | [Amount of Downloads]
+                Deprecated : True/False | Vulnerable : True/False
+                <Description of the Package(Readme file)>
+                License URL : [URL]
+
+#### **New Option `-exact-match`**
```

Comment:
This is a new command, so technically all options are new. I'd just list out the arguments/options and then have a section for each one to provide how they work.